app/controlplane/pkg/biz/group.go

```diff
@@ -355,86 +396,151 @@ func (uc *GroupUseCase) AddMemberToGroup(ctx context.Context, orgID uuid.UUID, o
+}
+
+// resolveAndValidateGroup resolves the group ID and verifies the group exists
+func (uc *GroupUseCase) resolveAndValidateGroup(ctx context.Context, orgID uuid.UUID, opts *AddMemberToGroupOpts) (uuid.UUID, *Group, error) {
+	resolvedGroupID, err := uc.ValidateGroupIdentifier(ctx, orgID, opts.ID, opts.Name)
+	if err != nil {
+		return uuid.Nil, nil, err
+	}
+
 	// Check the group exists
 	existingGroup, err := uc.groupRepo.FindByOrgAndID(ctx, orgID, resolvedGroupID)
 	if err != nil {
-		return nil, fmt.Errorf("failed to find group: %w", err)
+		return uuid.Nil, nil, fmt.Errorf("failed to find group: %w", err)
 	}
 
 	if existingGroup == nil {
-		return nil, NewErrNotFound("group")
+		return uuid.Nil, nil, NewErrNotFound("group")
 	}
 
+	return resolvedGroupID, existingGroup, nil
+}
+
+// validateRequesterPermissions checks if the requester has sufficient permissions
+func (uc *GroupUseCase) validateRequesterPermissions(ctx context.Context, orgID, requesterID, groupID uuid.UUID) error {
 	// Check if the requester is part of the organization
-	requesterMembership, err := uc.membershipRepo.FindByOrgAndUser(ctx, orgID, opts.RequesterID)
+	requesterMembership, err := uc.membershipRepo.FindByOrgAndUser(ctx, orgID, requesterID)
 	if err != nil && !IsNotFound(err) {
-		return nil, NewErrValidationStr("failed to check existing membership")
+		return NewErrValidationStr("failed to check existing membership")
 	}
 
 	if requesterMembership == nil {
-		return nil, NewErrValidationStr("requester is not a member of the organization")
+		return NewErrValidationStr("requester is not a member of the organization")
 	}
 
-	// Check if the requester has sufficient permissions
 	// Allow if the requester is an org owner or admin
 	isAdminOrOwner := requesterMembership.Role == authz.RoleOwner || requesterMembership.Role == authz.RoleAdmin
+	if isAdminOrOwner {
+		return nil
+	}
 
 	// If not an admin/owner, check if the requester is a maintainer of this group
-	if !isAdminOrOwner {
-		// Check if the requester is a maintainer of this group
-		requesterGroupMembership, err := uc.membershipRepo.FindByUserAndResourceID(ctx, opts.RequesterID, resolvedGroupID)
-		if err != nil && !IsNotFound(err) {
-			return nil, fmt.Errorf("failed to check requester's group membership: %w", err)
-		}
+	requesterGroupMembership, err := uc.membershipRepo.FindByUserAndResourceID(ctx, requesterID, groupID)
+	if err != nil && !IsNotFound(err) {
+		return fmt.Errorf("failed to check requester's group membership: %w", err)
+	}
 
-		// If not a maintainer of this group, deny access
-		if requesterGroupMembership == nil || requesterGroupMembership.Role != authz.RoleGroupMaintainer {
-			return nil, NewErrValidationStr("requester does not have permission to add members to this group")
-		}
+	// If not a maintainer of this group, deny access
+	if requesterGroupMembership == nil || requesterGroupMembership.Role != authz.RoleGroupMaintainer {
```

Comment:
Weren't we using the enforcer here?

Reply:
on the service layer, here we check it again just in case